app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java

```diff
@@ -20,33 +21,39 @@
     /**
      * Checks if all permissions are granted.
      *
-     * @param permissions The permission(s) to check. It can't be null.
-     * @return true if the app has all permissions granted, false otherwise.
+     * @param context     The target Context. It can't be null.
+     * @param permissions The permission(s) to check.
+     * @return false if the app has at least 1 permission denied, true otherwise.
      * @author Francesco Bau', helped by Alberto Ursino.
      */
-    public static boolean checkPermissions(Context context, @NonNull String[] permissions) {
-        for (String permission : permissions) {
-            if (isGranted(context, permission))
-                return false;
+    public static boolean checkPermissions(@NonNull Context context, @Nullable String[] permissions) {
+        if (permissions != null) {
+            for (String permission : permissions) {
+                if (isGranted(context, permission))
+                    return false;
+            }
         }
         return true;
     }
 
     /**
      * Checks which permissions are NOT granted.
      *
+     * @param context     The target Context. It can't be null.
      * @param permissions The permission(s) to check.
      * @return The NOT granted permission(s).
      * @author Francesco Bau'
      * @author Alberto Ursino
      */
-    public static String[] getDeniedPermissions(Context context, String[] permissions) {
+    public static String[] getDeniedPermissions(@NonNull Context context, @Nullable String[] permissions) {
         ArrayList<String> deniedPermissions = new ArrayList<>();
         int arrayLength = 0;
-        for (int i = 0; i < permissions.length; i++) {
-            if (isGranted(context, permissions[i])) {
-                deniedPermissions.add(permissions[i]);
-                arrayLength++;
+        if (permissions != null) {
+            for (int i = 0; i < permissions.length; i++) {
```

Comment:
Inside this method you use the classic for loop statement.
Use always the same!

Reply:
Oh, now I see what you mean. Thank you.
I'm gonna fix that right now.